tests/unit/Snak/SnakListTest.php

```diff
@@ -294,6 +295,50 @@ public function testOrderByProperty( SnakList $snakList, SnakList $expected, arr
 		}
 	}
 
+	public function testComparableInterface() {
+		$this->assertInstanceOf( Comparable::class, new SnakList() );
+	}
+
+	/**
+	 * @dataProvider equalsProvider
+	 */
+	public function testEquals( SnakList $list1, SnakList $list2, $expected ) {
+		$this->assertSame( $expected, $list1->equals( $list2 ) );
+	}
+
+	public function equalsProvider() {
+		$empty = new SnakList();
+		$oneSnak = new SnakList( [ new PropertyNoValueSnak( 1 ) ] );
+
+		return [
+			'empty object is equal to itself' => [
+				$empty,
+				$empty,
+				true
+			],
+			'non-empty object is equal to itself' => [
+				$oneSnak,
+				$oneSnak,
+				true
+			],
+			'different empty objects are equal' => [
+				$empty,
+				new SnakList(),
+				true
+			],
+			'different objects with same content are equal' => [
+				$oneSnak,
+				new SnakList( [ new PropertyNoValueSnak( 1 ) ] ),
+				true
+			],
+			'different objects with different content are not equal' => [
+				$oneSnak,
+				new SnakList( [ new PropertyNoValueSnak( 2 ) ] ),
+				false
```

Comment:
I'd rather see this as a separate test (as in separate method) instead of squashing both test case data, and the expected result in the data provider method but whatever.